hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/parser/HoodieCommonSqlParser.scala

```diff
@@ -57,6 +57,14 @@ class HoodieCommonSqlParser(session: SparkSession, delegate: ParserInterface)
 
   override def parseDataType(sqlText: String): DataType = delegate.parseDataType(sqlText)
 
+  /* SPARK-37266 Added parseQuery to ParserInterface in Spark 3.3.0. This is a patch to prevent
+   hackers from tampering text with persistent view, it won't be called in older Spark
+   Don't mark this as override for backward compatibility
+   Can't use sparkExtendedParser directly here due to the same reason */
```

Comment:
Sorry, but i can't understand the java-doc: can you please elaborate on why this is here? 

 - What exactly are we trying to prevent from happening? 
 - What BWC are we referring to?

Reply:
`parseQuery` is a new method of Spark trait `ParserInterface`. there would be compile issue If we call this method from any class that’s shared across different versions of spark, because older `ParserInterface` doesn’t have this method

Due to the same reason, we can't mark this method with `override` because for older spark there isn't `parseQuery`

Reply:
As discussed on Slack, let's instead of doing parsing in `SparkAdapter` create `ExtendedParserInterface`, where we can place this new `parseQuery` method and that could be used in Hudi's code-base (this is similar to how `HoodieCatalystExpressionUtils` set up)